checker-util/src/test/java/org/checkerframework/checker/regex/util/RegexUtilTest.java

```diff
@@ -50,5 +63,30 @@ public void test_isRegex_and_asRegex() {
         Assert.assertThrows(Error.class, () -> RegexUtil.asRegex(s4, 1));
         Assert.assertFalse(RegexUtil.isRegex(s4, 2));
         Assert.assertThrows(Error.class, () -> RegexUtil.asRegex(s4, 2));
+        Assert.assertFalse(RegexUtil.isRegex(s4, 1, 1));
+        Assert.assertThrows(Error.class, () -> RegexUtil.asRegex(s4, 1, 1));
+    }
+
+    @Test
+    public void test_getNonNullGroups() {
+        String s1 = "\\(abc\\)?(123)";
+        String s2 = "([(abc)])*(xyz)\\?(abc)";
+        String s3 = "(\\Q()()\\E)";
+        String s4 = "([abc\\Qwww\\E])(abc)?";
+        String s5 = "[(abc]";
+
+        Assert.assertThrows(Error.class, () -> RegexUtil.getNonNullGroups(s1, 2));
+        Assert.assertEquals(
+                new ArrayList<>(Collections.singletonList(1)), RegexUtil.getNonNullGroups(s1, 1));
```

Comment:
There is no need to use `new ArrayList` here and in the later lines of this file.